Q3:Sales of products in four diffrent regions is tabulated for males and females.Find if male-female buyer rations are similar across regions

In [ ]:
#Here we use chi-square Test

Chi-square test: Chi-square tests of independence test whether two qualitative variables are independent.That is whether there exist a relationship between two categorical variables.

Null hypothesis:The variables are independent,there is no dependency between the gender based buyer rations across regions

ie,Ho: The male-female buyer rations are similar across regions

Alternate hypothesis:The variables are dependent,there is significant dependency between the gender based buyer rations across regions.

ie,Ha:The male-female buyer rations are not similar across regions

In [1]:
import numpy as np
import pandas as pd
from scipy import stats

In [2]:
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


In [3]:
df=pd.read_csv("BuyerRatio.csv")

In [4]:
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [7]:
df_table=df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [8]:
df_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [9]:
value=stats.chi2_contingency(df_table)
value

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [10]:
#degrees of freedom=(no. of rows-1)*(no. of columns-1)
dof=value[2]
dof

3

In [11]:
Expected_value=value[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [12]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2/e for o,e in zip(df_table.values,Expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [13]:
# from chi-square table(α = 0.05 and df=3)
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

test statistic=1.515295 < critical_value=7.8147279\
failed to reject null hypothesis

ie,There is no dependency between male-female buyers rations and are similar across region.